cheri-c-programming.tex

```diff
@@ -1213,15 +1214,28 @@ \subsection{Restricting capability properties}
   new capability.
 
 \item[\ccode{void *cheri\_csetbounds(void *c, size\_t x)}] Narrow the bounds of capability
-  \cvar{c} so that the lower bound is the current pointer value (which may
+  \cvar{c} so that the lower bound is the current address (which may
   have been increased relative to \cvar{c}'s original lower bound), and its
   upper bound is suitable for a length of \cvar{x}.
-%  This macro wraps the compiler built-in
+
+  Note that the effective bounds of the returned capability may be
+  wider than the range [\ccode{cheri\_getaddress(c)},
+  \ccode{cheri\_getaddress(c) + x}) due to capability compression (see
+  \Cref{sec:bounds_alignment}).
```

Comment:
:+1:
Maybe add "... but the upper bound of the returned capability will never exceed the upper bound of \ccode{c}" ?

Reply:
I might go with the slightly more general statement that the new bounds will always be a subset of the existing bounds so it also covers lower bounds.